ted_sws/notice_validator/services/sparql_test_suite_runner.py

```diff
@@ -126,41 +125,49 @@ class SPARQLReportBuilder:
         Given a SPARQLQueryResult, generates JSON and HTML reports.
     """
 
-    def __init__(self, sparql_test_suite_execution: SPARQLTestSuiteValidationReport):
+    def __init__(self, sparql_test_suite_execution: SPARQLTestSuiteValidationReport, notice_id: List[str] = None):
         self.sparql_test_suite_execution = sparql_test_suite_execution
+        self.notice_id = notice_id
 
     def generate_report(self) -> SPARQLTestSuiteValidationReport:
-        html_report = TEMPLATES.get_template(SPARQL_TEST_SUITE_EXECUTION_HTML_REPORT_TEMPLATE).render(
-            self.sparql_test_suite_execution.dict())
+        tpl_data: dict = self.sparql_test_suite_execution.dict()
+        tpl_data["notice_id"] = self.notice_id
+        html_report = TEMPLATES.get_template(SPARQL_TEST_SUITE_EXECUTION_HTML_REPORT_TEMPLATE).render(tpl_data)
         self.sparql_test_suite_execution.object_data = html_report
         return self.sparql_test_suite_execution
 
 
-def validate_notice_with_sparql_suite(notice: Notice, mapping_suite_package: MappingSuite):
+def validate_notice_with_sparql_suite(notice: Union[Notice, List[Notice]], mapping_suite_package: MappingSuite):
     """
     Validates a notice with a sparql test suites
     :param notice:
     :param mapping_suite_package:
     :return:
     """
 
-    def sparql_validation(rdf_manifestation: RDFManifestation) -> List[SPARQLTestSuiteValidationReport]:
+    def sparql_validation(_notice: Notice) \
```

Comment:
In formal list of arguments don't use underscore (_) as a first character for arguments names.